src/nyaml/nxdl2nyaml.py

```diff
@@ -686,6 +686,15 @@ def handle_dim_with_value_and_index(depth, node, file_out, dimension_doc):
                             else f"({child_attrs['value']},)"
                         )
                         node.remove(child)
+                    # the next branch is only for legacy support of deprecated ref
+                    # as seen in NXdetector_group
+                    elif child_attrs.get("ref", "") and not dimension_doc:
+                        dim_index_value = (
+                            f"{dim_index_value[:-1]} {child_attrs['ref']},)"
+                            if dim_index_value
+                            else f"({child_attrs['ref']},)"
+                        )
+                        node.remove(child)
```

Comment:
I see that only one attribute either `value` or `ref` will be handled here. Both `ref` and `value` should be managed together. Also, the variable `dim_index_value` is being overwritten. Need another variable for `ref` e.g. `dim_ref` and the ref needs to be written in `yaml`

Reply:
I missed this function with_all_dim_attr function, as I followed the execution path of the debugger which documents that for the example NXdetector_group.nxdl.xml and converting to yaml, the function handle_dim_with_all_dim_attr is not called even not for <dimensions><dim index="1" ref="group_index"/></dimensions>, lemme check why.

Reply:
Root cause of the problem is different, see comment in the handle_dimensions_function lines797 and following [1cee1ed](https://github.com/FAIRmat-NFDI/nyaml/pull/47/commits/1cee1ed7fa1f584c76865fda4e1902f597dd59af) 
This should be fixed in an own refactoring to bugfix branch an PR that is merged in here and that fixes only this issue.

Reply:
I think the issue is in `NXdetector_group.nxdl.xml` which is not created properly. What is axis `index` without knowing the `vaue (axis name)`. We may consider raising error saying that `value` is required for an given `index`.

Reply:
NXdetector_group uses currently deprecated stuff, albeit this is not an argument as to why this deprecated stuff should not be parsed and convertered from NXDL to YAML correctly.

I am supportive though to the idea that if an NXDL uses incorrect syntax that the parser should throw.

Reply:
this can be checked during the implementation related to issue #52